File: lib/core/security/security_manager.dart
import 'package:shared_preferences/shared_preferences.dart';

class SecurityManager {
  SharedPreferences? _prefs;

  Future<void> init() async {
    _prefs = await SharedPreferences.getInstance();
  }

  bool get isBiometricsEnabled => _prefs?.getBool('biometrics_enabled') ?? false;
  set isBiometricsEnabled(bool value) => _prefs?.setBool('biometrics_enabled', value);

  bool get isPinEnabled => _prefs?.getBool('pin_enabled') ?? false;
  set isPinEnabled(bool value) => _prefs?.setBool('pin_enabled', value);

  String get userPin => _prefs?.getString('user_pin') ?? "";
  set userPin(String value) => _prefs?.setString('user_pin', value);

  bool get isCloudBackupEnabled => _prefs?.getBool('cloud_backup_enabled') ?? false;
  set isCloudBackupEnabled(bool value) => _prefs?.setBool('cloud_backup_enabled', value);

  String get baseCurrency => _prefs?.getString('base_currency') ?? "SAR";
  set baseCurrency(String value) => _prefs?.setString('base_currency', value);

  String get selectedLanguage => _prefs?.getString('selected_language') ?? "en";
  set selectedLanguage(String value) => _prefs?.setString('selected_language', value);

  bool verifyPin(String inputPin) => userPin == inputPin;
}

File: app/src/main/java/com/example/core/security/SecurityManager.kt
package com.example.core.security

import android.content.Context

class SecurityManager(context: Context) {
    private val prefs = context.getSharedPreferences("pfms_security_prefs", Context.MODE_PRIVATE)

    var isBiometricsEnabled: Boolean
        get() = prefs.getBoolean("biometrics_enabled", false)
        set(value) = prefs.edit().putBoolean("biometrics_enabled", value).apply()

    var isPinEnabled: Boolean
        get() = prefs.getBoolean("pin_enabled", false)
        set(value) = prefs.edit().putBoolean("pin_enabled", value).apply()

    var userPin: String
        get() = prefs.getString("user_pin", "") ?: ""
        set(value) = prefs.edit().putString("user_pin", value).apply()

    var isCloudBackupEnabled: Boolean
        get() = prefs.getBoolean("cloud_backup_enabled", false)
        set(value) = prefs.edit().putBoolean("cloud_backup_enabled", value).apply()

    var baseCurrency: String
        get() = prefs.getString("base_currency", "SAR") ?: "SAR"
        set(value) = prefs.edit().putString("base_currency", value).apply()

    var selectedLanguage: String
        get() = prefs.getString("selected_language", "en") ?: "en"
        set(value) = prefs.edit().putString("selected_language", value).apply()

    fun verifyPin(inputPin: String): Boolean {
        return userPin == inputPin
    }
}
